R/derive_params_growth_age.R

```diff
@@ -239,9 +253,13 @@ derive_params_growth_age <- function(dataset,
       relationship = "many-to-many"
     ) %>%
     mutate(age_diff := abs(metadata_age - {{ age }})) %>%
-    group_by(USUBJID) %>%
+    group_by(!!!by_vars) %>%
     mutate(is_lowest = age_diff == min(age_diff)) %>%
-    filter(is_lowest)
+    ungroup() %>%
+    group_by(!!!by_vars, is_lowest) %>%
+    filter(is_lowest & row_number() == 1) %>%
+    ungroup()
```

Comment:
@Fanny-Gautier please can you double check this block of code. I trust your R skills more than mine to see that its robust and an efficient way to achieve this. Let us know if you spot any improvements.

Reply:
There is an issue with the new code : it does not keep all the data/visits as expected. I guess it comes from the filter with row_number() == 1. If you group by STUDYID, USUBJID only, then it keeps only the first record (e.g. VISIT ="Screening 1") per patient.
![image](https://github.com/pharmaverse/admiralpeds/assets/157114584/36cace84-4b00-4b62-8045-5ce9a4e2803b)

As previously we should have here 246 obs :
![image](https://github.com/pharmaverse/admiralpeds/assets/157114584/769fc7a3-c659-4555-8c99-d0cfa8263ad9)

Reply:
sometimes `is_lowest` will flag 2 records for 1 subject if it is _exactly_ down the middle between two values, because it is sorted prior, it will take the "lower" one, 

for example, 
current age = 25 months
metadata has records for 24.5 and 25.5 months
the record chosen for the match will be the 24.5 

I am seeing some issues though, looking into it today

Reply:
Okay I think I figured it out, with the newest implementation of the `is_lowest` logic, I actually need the `VISIT` information in the `by_vars` call as well @Fanny-Gautier, please take another look at your latest convenience, otherwise, I've adopted all other feedback I think too

Reply:
It's corrected, I get the right results in the ADVS template. thanks for the update.